Meta/serenity.sh

```diff
@@ -90,6 +90,19 @@ fi
 
 CMAKE_ARGS=()
 
+NPROC="nproc"
+SYSTEM_NAME="$(uname -s)"
+
+if [ "$SYSTEM_NAME" = "OpenBSD" ]; then
+    NPROC="sysctl -n hw.ncpuonline"
+elif [ "$SYSTEM_NAME" = "FreeBSD" ]; then
+    NPROC="sysctl -n hw.ncpu"
+elif [ "$SYSTEM_NAME" = "Darwin" ]; then
+    NPROC="sysctl -n hw.ncpu"
+fi
+
+[ -z "$MAKEJOBS" ] && MAKEJOBS=$($NPROC)
```

Comment:
I stole this from BuildMold but some of the other script have simpler variants. Is this alright?

Reply:
```suggestion
MAKEJOBS=$(cmake -P processor-count.cmake 2>&1)
```

Where `processor-count.cmake` is just:

```cmake
include(ProcessorCount)
ProcessorCount(N)
message("${N}")
```

I think that's simpler.